tests/unit_tests/core/tuple_accessor.cpp

```diff
@@ -0,0 +1,41 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <iostream>
+
+#include "detray/definitions/detail/accessor.hpp"
+
+// Thrust include(s).
+#include <thrust/tuple.h>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+TEST(tuple_accessor, tuple_accessor) {
+
+    using namespace detray;
+
+    // std::tuple test
+    auto s_tuple = detail::make_tuple<std::tuple>(1.0, 2, "std::tuple");
+
+    const int s_tuple_size = detail::tuple_size<decltype(s_tuple)>::value;
+
+    EXPECT_EQ(s_tuple_size, 3);
+    EXPECT_EQ(detail::get<0>(s_tuple), 1.0);
```

Comment:
This should work... But I just always fear equality checks for floating point numbers. I think `EXPECT_FLOAT_EQ(...)` would be a better macro to use here.

https://github.com/google/googletest/blob/main/googletest/include/gtest/gtest.h#L1965

Reply:
OK. Didn't know that `EXPECT_FLOAT_EQ` exists..